packages/perseus/src/util/parse-perseus-json/general-purpose-parsers/constant.ts

```diff
@@ -0,0 +1,12 @@
+import type {Parser} from "../parser-types";
+
+type Primitive = string | number | boolean | null | undefined;
+
+export function constant<T extends Primitive>(acceptedValue: T): Parser<T> {
+    return (rawValue, ctx) => {
+        if (rawValue !== acceptedValue) {
+            return ctx.failure(JSON.stringify(acceptedValue), rawValue);
+        }
+        return ctx.success(acceptedValue);
+    };
+}
```

Comment:
Possible bug: `JSON.stringify(undefined)` returns undefined, not a string. I think this might result in the correct overall behavior anyway, but it's not, as Adam Savage would say, a "positive fit".